### PR TITLE
feat: add 'kpm import' to converts other formats to KCL file

### PR DESCRIPTION
**Pull request by jakezhu9**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

#192

#### 2. What is the scope of this PR (e.g. component or file name):

pkg/cmd

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

image

### CODE REVIEW

**Comment by zong-zhe** [comment]
fix the e2e test case in https://github.com/kcl-lang/kpm/blob/main/test/e2e/test_suites/kpm/exec_outside_pkg/help_msg/test_suite.stdout. Add the command 'import' msg         into it.

**Comment by jakezhu9** [comment]
> fix the e2e test case in https://github.com/kcl-lang/kpm/blob/main/test/e2e/test_suites/kpm/exec_outside_pkg/help_msg/test_suite.stdout. Add the command 'import' msg into it.

done

**Comment by jakezhu9** [comment]
@Peefy Could you please take a look at this?

File: pkg/cmd/cmd_import.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import (
+	"github.com/urfave/cli/v2"
+	"kcl-lang.io/kcl-go/pkg/tools/gen"
+	"kcl-lang.io/kpm/pkg/client"
+	"kcl-lang.io/kpm/pkg/reporter"
+	"os"
+)
+
+// NewImportCmd new a Command for `kpm import`.
+func NewImportCmd(_ *client.KpmClient) *cli.Command {
```
Comment (zong-zhe) [inline pkg/cmd/cmd_import.go:14]: `KpmClient` is useful and is mainly responsible for control all functions provided in the kpm cli.

File: pkg/cmd/cmd_import.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+			},
+			&cli.BoolFlag{
+				Name:    "force",
+				Aliases: []string{"f"},
+				Usage:   "force overwrite output file",
+			},
+		},
+		Action: func(c *cli.Context) error {
+			if c.Args().Len() != 1 {
+				reporter.Report("kpm: invalid arguments")
```
Comment (zong-zhe) [inline pkg/cmd/cmd_import.go:50]: Use `reporter.ReportMsgTo` to entrue that `KpmClient` can control the output message.

```
reporter.ReportMsgTo("kpm: invalid arguments", kpmcli.GetLogWriter())
```

`kpmcli` is the parameter of `NewImportCmd(kpmcli *client.KpmClient)`.

File: pkg/cmd/cmd_import.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+			}
+
+			outputFile := c.String("output")
+			if outputFile == "" {
+				outputFile = "generated.k"
+				reporter.Report("kpm: output file not specified, use default: ", outputFile)
+			}
+
+			if _, err := os.Stat(outputFile); err == nil && !c.Bool("force") {
+				reporter.ExitWithReport("kpm: output file already exist, use --force to overwrite: ", outputFile)
```
Comment (zong-zhe) [inline pkg/cmd/cmd_import.go:81]: Return an error with the error message.
```
return fmt.Errorf("kpm: output file already exist, use --force to overwrite: ", outputFile)
```

File: pkg/cmd/cmd_import.go (hunk 2 above)
Comment (jakezhu9) [reply pkg/cmd/cmd_import.go]: 👌

File: pkg/cmd/cmd_import.go (hunk 3 above)
Comment (jakezhu9) [reply pkg/cmd/cmd_import.go]: fixed

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**